packages/swing-store/src/bundleStore.js

```diff
@@ -72,14 +74,25 @@ export function makeBundleStore(db, ensureTxn, noteExport = () => {}) {
    * @param {Bundle} bundle
    */
   function addBundle(bundleID, bundle) {
-    const { moduleFormat } = bundle;
-    if (moduleFormat !== 'endoZipBase64')
-      throw Fail`unsupported module format ${q(moduleFormat)}`;
-    const { endoZipBase64, endoZipBase64Sha512 } = bundle;
-    bundleID === bundleIdFromHash(endoZipBase64Sha512) ||
-      Fail`bundleID ${q(bundleID)} does not match bundle`;
     ensureTxn();
-    sqlAddBundle.run(bundleID, decodeBase64(endoZipBase64));
+    const { moduleFormat } = bundle;
+    if (bundleID.startsWith('b0-')) {
+      if (moduleFormat !== 'nestedEvaluate')
+        throw Fail`unsupported b0- module format ${q(moduleFormat)}`;
+      const serialized = JSON.stringify(bundle);
+      if (bundleID !== bundleIdFromHash(0, createSHA256(serialized).finish()))
+        throw Fail`bundleID ${q(bundleID)} does not match bundle`;
```

Comment:
Since we're not trying to gate via type info here, the weird `if (...) throw Fail...` construct isn't needed here.

Reply:
True, but I'm finding that a line with `if` and a line with `throw` or `Fail` is easier to read as a security check than a line with a short-circuiting `||`. And the presence of `throw Fail` two lines earlier (needed for the typechecker) then makes its presence in the security check more plausible.

Of these options, which feels the best?:

```js
      if (bundleID !== bundleIdFromHash(0, createSHA256(serialized).finish())) {
        throw Error(`bundleID ${q(bundleID)} does not match bundle`);
     }
```

```js
      if (bundleID !== bundleIdFromHash(0, createSHA256(serialized).finish())) {
        Fail`bundleID ${q(bundleID)} does not match bundle`;
     }
```

```js
      bundleID !== bundleIdFromHash(0, createSHA256(serialized).finish()) ||
        Fail`bundleID ${q(bundleID)} does not match bundle`;
```

Reply:
ah, and see, I already messed up the short-circuiting example, the `!==` needs to be a `===`

Reply:
I'm going to go with `throw Error`, I think that's pretty clear.